vmware-tanzu/helm-charts
# ci: introduce CI workflow to verify the CRD folder

#### Special notes for your reviewer:
Close #480 

Introduce CI workflow to verify the CRD folder, this will do following:
- Set up Github Action environment: checkout code, setup kind, velero
- Use `velero` and `kubectl `command to generate 2 YAML manifest files
- Use diff command to check if these files are in synced or not
- Archive the diff workspace for later debugging

This was tested on my forked repo:
- case diff found: https://github.com/tungbq/helm-charts-velero/actions/runs/5751960913/job/15591857978
- case no diff found: https://github.com/tungbq/helm-charts-velero/actions/runs/5751904132/job/15591689104

The workflow also collects the artifact YAML manifest files and the diff results:
- For example: https://github.com/tungbq/helm-charts-velero/suites/14794943841/artifacts/841831563
- ![image](https://github.com/vmware-tanzu/helm-charts/assets/85242618/91afe4b6-f928-402c-a307-d6ed1b4c7fa9)

Currently I set the trigger event to run on `workflow_dispatch `only. Let me know if we need to cover other trigger events based.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Variables are documented in the values.yaml or README.md
- [ ] Title of the PR starts with chart name (e.g. `[velero]`)
